src/globus_cli/commands/gcs/endpoint/_common.py

```diff
@@ -0,0 +1,25 @@
+from globus_cli.termio import Field, formatters
+
+# https://docs.globus.org/globus-connect-server/v5.4/api/schemas/Endpoint_1_2_0_schema/
+GCS_ENDPOINT_FIELDS = [
+    Field("Endpoint ID", "id"),
+    Field("Display Name", "display_name"),
+    Field("Allow UDT", "allow_udt", formatter=formatters.FuzzyBool),
+    Field("Contact Email", "contact_email"),
+    Field("Contact Info", "contact_info"),
+    Field("Department", "department"),
+    Field("Description", "description"),
+    Field("Earliest Last Access", "earliest_last_access", formatter=formatters.Date),
+    Field("GCS Manager URL", "gcs_manager_url"),
+    Field("GridFTP Control Channel Port", "gridftp_control_channel_port"),
+    Field("Info Link", "info_link"),
+    Field("Keywords", "keywords"),
+    Field("Max Concurrency", "max_concurrency"),
+    Field("Max Parallelism", "max_parallelism"),
+    Field("Network Use", "network_use"),
```

Comment:
Can we do some custom format work for the network use parameters, since concurrency and parallelism are only meaningful if `network_use='custom'`?

We have four other parameters here, so I'm imagining something like...
```
Network Use:               custom
Network Use (concurrency): prefer=8 max=32
Network Use (parallelism): prefer=16 max=16
```
as well as
```
Network Use:               aggressive
Network Use (concurrency): aggressive
Network Use (parallelism): aggressive
```

Reply:
I'm not sure what I think about that suggestion.

It definitely looks better than the current version. Is the improved view simplicity sufficiently worth munging the response structure like that?

Reply:
> Is the improved view simplicity sufficiently worth munging the response structure like that?

I'm not sure if you mean "is the improvement worth it because we're supposed to reflect API response data" or "is it worth the effort/complexity for us"?

I don't think there is or should be any expectation that CLI text-mode outputs directly reflect API response data. We're already doing "work" to responses in a lot of cases, and I think this one isn't particularly much different from many others (though I can see an argument about this).

As for whether or not it's worth the complexity in terms of effort, I don't think it's that hard, if we get a little clever.
`Field` does its key lookup using `jmespath`, so I think we need a custom format and then the fields become something like
```python
Field(
    "Network Use (concurrency)",
    "[prefer_concurrency, max_concurrency]",
    formatter=FancyFormatter()
),
Field(
    "Network Use (parallelism)",
    "[prefer_parallelism, max_parallelism]",
    formatter=FancyFormatter()
),
```

Modulo whatever parts of that I got subtly wrong.

Reply:
I also love writing code! :grin: 

I think this is within bounds for us transforming outputs. I think that there are so many fields in the GCS output that the majority of users will benefit from us compacting things a bit, and we always have the crutch of `--format JSON` for folks who want the "real data".

Reply:
Sure, I agree more than I disagree.

Plus now I get to write more of that code stuff!

Reply:
![image](https://github.com/globus/globus-cli/assets/113056046/85b9ad24-2180-4482-b695-93d55afb51a5)

Reply:
Alright, here's what it looks like

```
globus-cli % globus gcs endpoint show $EP
Endpoint ID:                  379f7156-c996-4de5-a228-db2a4d872f89
Display Name:                 Dereks Cool GCSv5.4 Sandbox Endpoint
Allow UDT:                    False
Contact Email:                derek@globus.org
Contact Info:                 401 North Michigan Ave
Department:                   Automate
Description:                  This is my endpoint. There are many like it but this one is mine.
Earliest Last Access:         2023-12-15
GCS Manager URL:              https://0629c.5898.sandbox2.zones.dnsteam.globuscs.info
GridFTP Control Channel Port: 443
Info Link:                    None
Keywords:                     Derek,Derk,MyCoolEndpoint
Network Use:                  custom
Network Use (Concurrency):    Preferred: 3, Max: 5
Network Use (Parallelism):    Preferred: 2, Max: 8
Organization:                 Globus
Public:                       True
Subscription ID:              4fa609fc-14f6-4c62-acfd-680873d3b6fe
```

```
globus-cli % globus gcs endpoint update $EP --network-use aggressive
Endpoint ID:                  379f7156-c996-4de5-a228-db2a4d872f89
Display Name:                 Dereks Cool GCSv5.4 Sandbox Endpoint
Allow UDT:                    False
Contact Email:                derek@globus.org
Contact Info:                 401 North Michigan Ave
Department:                   Automate
Description:                  This is my endpoint. There are many like it but this one is mine.
Earliest Last Access:         2023-12-15
GCS Manager URL:              https://0629c.5898.sandbox2.zones.dnsteam.globuscs.info
GridFTP Control Channel Port: 443
Info Link:                    None
Keywords:                     Derek,Derk,MyCoolEndpoint
Network Use:                  aggressive
Network Use (Concurrency):    aggressive
Network Use (Parallelism):    aggressive
Organization:                 Globus
Public:                       True
Subscription ID:              4fa609fc-14f6-4c62-acfd-680873d3b6fe

```

Reply:
> Keywords:                     Derek,Derk,MyCoolEndpoint

This perfectly mirrors my own endpoints.